tiledb/sm/array/array.cc

```diff
@@ -854,9 +854,12 @@ uint64_t Array::timestamp_end_opened_at() const {
   return timestamp_end_opened_at_;
 }
 
-Status Array::set_config(Config config) {
+void Array::set_config(Config config) {
+  if (is_opening_or_closing_) {
```

Comment:
If the array is open already then I believe set_config should have no effect, except for reopen calls (which we'll have to figure out how to deal with).